catroid/src/main/java/org/catrobat/catroid/transfers/FacebookLoginHandler.java

```diff
@@ -0,0 +1,164 @@
+ */
+
+package org.catrobat.catroid.transfers;
+
+import android.app.Activity;
+import android.content.SharedPreferences;
+import android.os.Bundle;
+import android.preference.PreferenceManager;
+
+import com.facebook.AccessToken;
+import com.facebook.FacebookCallback;
+import com.facebook.FacebookException;
+import com.facebook.login.LoginResult;
+
+import org.catrobat.catroid.common.Constants;
+import org.catrobat.catroid.ui.recyclerview.dialog.login.OAuthUsernameDialogFragment;
+import org.catrobat.catroid.ui.recyclerview.dialog.login.SignInCompleteListener;
+import org.catrobat.catroid.utils.UtilDeviceInfo;
+
+public class FacebookLoginHandler implements FacebookCallback<LoginResult>,
+		GetFacebookUserInfoTask.OnGetFacebookUserInfoTaskCompleteListener,
+		CheckOAuthTokenTask.OnCheckOAuthTokenCompleteListener,
+		FacebookLogInTask.OnFacebookLogInCompleteListener,
+		FacebookExchangeTokenTask.OnFacebookExchangeTokenCompleteListener,
+		CheckEmailAvailableTask.OnCheckEmailAvailableCompleteListener {
+
+	Activity activity;
+
+	public static final String FACEBOOK_PROFILE_PERMISSION = "public_profile";
+	public static final String FACEBOOK_EMAIL_PERMISSION = "email";
+
+	public FacebookLoginHandler(Activity activity) {
+		this.activity = activity;
+	}
+
+	@Override
+	public void onSuccess(LoginResult loginResult) {
+		AccessToken accessToken = loginResult.getAccessToken();
+		GetFacebookUserInfoTask getFacebookUserInfoTask = new GetFacebookUserInfoTask(activity, accessToken.getToken(),
+				accessToken.getUserId());
+		getFacebookUserInfoTask.setOnGetFacebookUserInfoTaskCompleteListener(this);
+		getFacebookUserInfoTask.execute();
+	}
+
+	@Override
+	public void onCancel() {
+		((SignInCompleteListener) activity).onLoginCancel();
+	}
+
+	@Override
+	public void onError(FacebookException error) {
+		((SignInCompleteListener) activity).onLoginCancel();
+	}
+
+	@Override
+	public void onGetFacebookUserInfoTaskComplete(String id, String name, String locale, String email) {
+		SharedPreferences sharedPreferences = PreferenceManager.getDefaultSharedPreferences(activity);
+		sharedPreferences.edit().putString(Constants.FACEBOOK_ID, id).commit();
+		sharedPreferences.edit().putString(Constants.FACEBOOK_USERNAME, name).commit();
+		sharedPreferences.edit().putString(Constants.FACEBOOK_LOCALE, locale).commit();
+
+		//if user has approved email permission, fb-email address is taken, else device email address
+		if (email != null) {
+			sharedPreferences.edit().putString(Constants.FACEBOOK_EMAIL, email).commit();
+		} else {
+			sharedPreferences.edit().putString(Constants.FACEBOOK_EMAIL,
+					UtilDeviceInfo.getUserEmail(activity)).commit();
+		}
+
+		CheckOAuthTokenTask checkOAuthTokenTask = new CheckOAuthTokenTask(activity, id, Constants.FACEBOOK);
+		checkOAuthTokenTask.setOnCheckOAuthTokenCompleteListener(this);
+		checkOAuthTokenTask.execute();
+	}
+
+	@Override
+	public void forceSignIn() {
+		//TODO: Show SignInDialog again.
```

Comment:
maybe we should finish this first?

Reply:
yeah, this kind of doesnt do anything anymore :-) since we dont have any sign in dialog anymore, ill just display a toast....